tests/integration_tests/test_keyfile.py

```diff
@@ -9,126 +9,151 @@
-    from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
-    from cryptography.fernet import Fernet
-    from cryptography.hazmat.primitives import hashes
-    from cryptography.hazmat.backends import default_backend
-
-    __SALT = b"Iguesscyborgslikemyselfhaveatendencytobeparanoidaboutourorigins"
-    
-    def __generate_key(password):
-        kdf = PBKDF2HMAC(algorithm=hashes.SHA256(), salt=__SALT, length=32, iterations=10000000, backend=default_backend())
-        key = base64.urlsafe_b64encode(kdf.derive(password.encode()))
-        return key
-
-    pw = 'fakepasssword238947239'
-    data = b'encrypt me!'
-    key = __generate_key(pw)
-    cipher_suite = Fernet(key)
-    encrypted_data = cipher_suite.encrypt(data)
-
-    decrypted_data = decrypt_keyfile_data( encrypted_data, pw)
-    assert decrypted_data == data
-
-def test_user_interface():
-    from bittensor._keyfile.keyfile_impl import ask_password_to_encrypt
-
-    with mock.patch('getpass.getpass', side_effect = ['pass', 'password', 'asdury3294y', 'asdury3294y']):
-        assert ask_password_to_encrypt() == 'asdury3294y'
-
-def test_overwriting():    
-    from bittensor._keyfile.keyfile_impl import KeyFileError
-    
-    keyfile = bittensor.keyfile (path = '/tmp/pytest/keyfile' )
-    alice = bittensor.Keypair.create_from_uri ('/Alice')
-    keyfile.set_keypair(alice, encrypt=True, overwrite=True, password = 'thisisafakepassword')
-    bob = bittensor.Keypair.create_from_uri ('/Bob')
-    
-    with pytest.raises(KeyFileError) as pytest_wrapped_e:
-        with mock.patch('builtins.input', return_value = 'n'):
-            keyfile.set_keypair(bob, encrypt=True, overwrite=False, password = 'thisisafakepassword')
-
-def test_keyfile_mock():
-    file = bittensor.keyfile( _mock = True )
-    assert file.exists_on_device()
-    assert not file.is_encrypted()
-    assert file.is_readable()
-    assert file.data
-    assert file.keypair
-    file.set_keypair( keypair = bittensor.Keypair.create_from_mnemonic( mnemonic = bittensor.Keypair.generate_mnemonic() ))
-
-def test_keyfile_mock_func():
-    file = bittensor.keyfile.mock()
+import bittensor
+
+
+class TestKeyFiles(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.root_path = f"/tmp/pytest{time.time()}"
+        os.makedirs(self.root_path)
+
+        self.create_keyfile()
+
+    def tearDown(self) -> None:
+        shutil.rmtree(self.root_path)
+
+    def create_keyfile(self):
+        keyfile = bittensor.keyfile(path=os.path.join(self.root_path, "keyfile"))
+
+        mnemonic = bittensor.Keypair.generate_mnemonic(12)
+        alice = bittensor.Keypair.create_from_mnemonic(mnemonic)
+        keyfile.set_keypair(alice, encrypt=True, overwrite=True, password='thisisafakepassword')
+
+        bob = bittensor.Keypair.create_from_uri('/Bob')
+        keyfile.set_keypair(bob, encrypt=True, overwrite=True, password='thisisafakepassword')
+
+        return keyfile
```

Comment:
This highlighted section is where most of the main changes are. The rest is a result of having converted to an OO style to facilitate using setUp/tearDown